### ISSUE TITLE
Cannot retrieve resource-translated message

### ISSUE THREAD

**Issue by enrico-padovani** [IntertechInc/aa-validation-from-server#2]
If I do set an annotation like this: [Range(1, short.MaxValue, ErrorMessageResourceType = typeof (Resources), ErrorMessageResourceName = "MyTranslatedResource")] I've got a null reference exception. I had a quick look on the code and the problem seems that is getting hard coded messages only and falls back into constants if a message is not provided.

Is there a chance that you can implement a resource-based error message? It will be great to have also the Display attribute translation if provided as field name.

I really like the integration idea and the reuse of validations between server and client. You did a nice job.

Enrico

**Comment by richinator38** [comment]
Yes, nice catch Enrico!  I did leave that out.  I can look at that and will let you know when done.  Thanks for the feedback!

**Comment by enrico-padovani** [comment]
Hi, what a fast answer!

So just to summarize the cases:
1. Annotation validation with no ErrorMessage: a standard message in English is provided and is used the property name Display in it **(handled)**
2. Annotation validation with an hard coded ErrorMessage: the message is shown as it is with no property name **(handled)**
3. Annotation validation attached to resource translation using ErrorMessageResourceType  and ErrorMessageResourceName: the messages is taken from the resources as it is with no property name **(new case)**

further enhancements:
the 1st feature could be extended by adding standard messages translated that overrides the constants provided by DataAnnotationConstants. Maybe by inject them into ValidationHelper.GetValidations() as an other overload.

I'll be happy to help you testing it!

**Comment by richinator38** [comment]
Ok, Enrico, it's ready for you to test.  I didn't do the further enhancement yet.

**Comment by enrico-padovani** [comment]
Hi there!

I've checked it with my project and since my Resources.resx was set Access Modifier: Public I've got always a null value message back (and fell back to the default). So I've changed line 107 of BaseValidationConverter.cs

from: var resName = rtype.GetProperty(resourceName, BindingFlags.NonPublic | BindingFlags.Static);

to: var resName = rtype.GetProperty(resourceName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);

and now it seems to work as expected.

I'll make some more tests but for now the overload of GetValidations that accept GetValidationsParms looks good.

Thanks a lot for the job done!
